### DISCUSSION TITLE
HC-tree SQLite and possibility of support for PocketBase

### DISCUSSION THREAD

**Discussion by katriellucas** [pocketbase/pocketbase#4253] [Q&A] [answered]
First things first, if you don't know what hctree is, please take a look at the [sqlite website](https://sqlite.org/hctree/doc/hctree/doc/hctree/index.html) to know more, specially the [tests page](https://sqlite.org/hctree/doc/hctree/doc/hctree/threadtest.wiki). In short, the project is an attempt to:

- Improve concurrency
- Support for replication
- Removal of database size limitations

This is just a question; the fork is active, though it doesn't seem stable yet, but I'm curious about the feasibility of this being implemented in PocketBase, as this seems to solve most of the writing problems for SQLite. I'm no master of PB internals, which is why I'm asking. Would this just be a drop-in replacement? Are there technological challenges that makes this hard to implement in the future?

**Comment by ganigeorgiev** [comment, marked as answer]
I've tested the hctree trunk a year ago and I think it is a really interesting experiment but it is not suitable for stable workloads (_the last time I tested it there was no recovery from power failure or process crash and you could end up with a corrupted db file_). Also the author of the implementation I remember mentioning that it probably will never be merged in the mainline SQLite source as it differs too much from how the SQLite "backend" (the blue part in https://www.sqlite.org/arch.html) operates.

With that said, in the near future there will be support for specifying a custom SQLite builds (as long as they are compatible with std `database/sql` driver; _technically users will be able to define connections to other databases like PostgreSQL, but at least until v1 I'm not planning officially supporting other SQL dialects_).

**Reply by katriellucas** [reply]
I see, interesting, so it's not completely out of question, but as figured, not stable yet.
